app/controllers/api/v1/participants_controller.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require "csv"
+
+module Api
+  module V1
+    class ParticipantsController < Api::ApiController
+      include ApiTokenAuthenticatable
+      include Pagy::Backend
+
+      before_action :ensure_lead_provider
+
+      def index
+        respond_to do |format|
+          format.json do
+            participant_hash = ParticipantSerializer.new(paginate(participants)).serializable_hash
+            render json: participant_hash.to_json
+          end
+          format.csv do
+            participant_hash = ParticipantSerializer.new(participants).serializable_hash
+            render body: to_csv(participant_hash)
+          end
+        end
+      end
+
+    private
+
+      def ensure_lead_provider
+        head :forbidden unless current_user.class == LeadProvider
+      end
+
+      def to_csv(hash)
+        return "" if hash[:data].empty?
+
+        headers = %w[id]
+        attributes = hash[:data].first[:attributes].keys
```

Comment:
It feels a bit messy having to do this despite having _a_ serializer, but I can't think of an alternative. Can you? 🤷

Reply:
Like, I feel like we should just be able to `to_csv` something.

Reply:
I had a quick look & think, but couldn't come up with anything. Reluctant to spend too much time on this given we're hoping to rip this out soon